hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -1023,6 +1023,9 @@ def as_strategy(strat_or_callable, thing, final=True):
                 "strings."
             )
         raise InvalidArgument(f"thing={thing!r} must be a type")  # pragma: no cover
+    if thing in types.TypeAliasTypes:  # pragma: no cover
+        # Code like `st.from_type(TypeAlias)` does not make sense.
+        raise InvalidArgument(f"thing={thing!r} does not make sense as a strategy")
```

Comment:
Maybe a message like "Cannot resolve TypeAlias to a strategy, because there are no instances of it at runtime"?